pysat/instruments/pysat_testing2d_xarray.py

```diff
@@ -120,3 +136,17 @@ def download(date_array, tag, sat_id, data_path=None, user=None, password=None):
 alt_profile_meta['density'] = {'units':'', 'long_name':'profiles'}
 alt_profile_meta['fraction'] = {'units':'', 'long_name':'profiles'}
 meta['alt_profiles'] = {'meta':alt_profile_meta, 'units':'', 'long_name':'profiles'}
+
+
+
+data, meta = load(r'C:\Data\Gayatri\US\UTD\SORTIE\UCol Summer Internship\Data\tiegcm_icon_merg2.0_totTgcm.s_149.nc')
```

Comment:
Filename will need to be provided a different way.

Reply:
This loading should be in its own routine. The testing2d_xarray is intended as a general testing Instrument object. We need a file labeled, say ucar_tiegcm.py located in pysat/instruments. You can use the netcdf_pandas routine as a starting point. You'll need to replace the call to load_netcdf4 in the load routine there with an xarray load call. The filename to be loaded will be provided by pysat in the fnames variable. You'll also have to set pandas_format = False at the top-level. The same command is in this testing file.

You'll also need to provide a template string for how the tiegcm filenames are formatted. Normally observational files have the date and/or time in the filename. In this case, to start things off you'll need to manually update the TIEGCM name to include a date.

Reply:
Gayatri, I'd like you to start on the individual TIEGCM instrument support file first, and then we'll iterate through the other changes you have developed once we have a better idea of where you are coming from.